templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -479,7 +479,7 @@ double IKFastKinematicsPlugin::harmonize(const std::vector<double> &ik_seed_stat
     while(solution[i] < 2*M_PI) {
       solution[i] += 2*M_PI;
     }
-    dist_sqr += fabs(ik_seed_state[i] - solution[i]);
+    dist_sqr += fabs(ss[i] - solution[i]);
```

Comment:
Good catch on this one.  I think the while loops arranged in this manner will likely result in joint values outside of the intended range [-2*PI , 2*PI] . The second while loop will always add 2*PI thus undoing the last iteration of the preceding while loop.  Is the real intent of this method to limit to joint range or could the objective be something else?